lib/include/openamp/virtio.h

```diff
@@ -120,6 +121,44 @@ struct virtio_device_id {
 #define VIRTIO_TRANSPORT_F_START      28
 #define VIRTIO_TRANSPORT_F_END        32
 
+#ifdef VIRTIO_DEBUG
+#include <metal/log.h>
+
+#define VIRTIO_ASSERT(_exp, _msg) do { \
```

Comment:
to fix checkpatch complain :

```
MACRO_ARG_REUSE: Macro argument reuse '_exp' - possible side-effects?
File:lib/include/openamp/virtio.h

```
you can use a local variable

```
#define VIRTIO_ASSERT(_exp, _msg) do ({ int __exp = _exp; \
		if (!(__exp)) { \
			metal_log(METAL_LOG_EMERGENCY, \
				  "FATAL: %s - " _msg, __func__); \
			metal_assert(__exp); \
		} \
	}
```

Same logic  for VIRTIO_RING_SIZE

Reply:
I used your suggestion for VIRTIO_ASSERT. This is not possible for VIRTIO_RING_SIZE, since it it reused by VRING_DECLARE.

Reply:
Not blocking , ok to ignore checkpatch complain on VIRTIO_RING_SIZE